### PR TITLE
Add SpyObj<T> typing for createSpyObj.

### PR DESCRIPTION
**Pull request by gmoothart**
Adding a mapped type SpyObj<T>, and make it the return value of
createSpyObj.

This update makes the return value of createSpyObj more flexible; You can use it as the original type but you can also treat the members as Spys without having to cast to jasmine.Spy.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dslint/dt.json" }`.

### CODE REVIEW

File: types/jasmine/index.d.ts [hunk 1]
```diff
@@ -514,6 +514,10 @@ declare namespace jasmine {
         argsForCall: any[];
     }
 
+    type SpyObj<T> = T & {
```
Comment (mhegazy) [inline types/jasmine/index.d.ts]: the top level intersection is sufficient. so: 
`T &  { [k in keyof T]: Spy; }`

Comment (gmoothart) [inline types/jasmine/index.d.ts]: Done, thanks.

**Comment by gmoothart** [comment]
ping - any other feedback?

**Comment by lukas-zech-software** [comment]
looks fine fore me :+1:

**Pull request merged**